pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/fluid/FluidManager.kt

```diff
@@ -0,0 +1,404 @@
+            }
+        }
+        return requestedFluids
+    }
+
+    private fun tick(segment: UUID) {
+        val suppliedFluids = getSuppliedFluids(segment)
+        val requestedFluids = getRequestedFluids(segment)
+
+        for ((fluid, suppliers) in suppliedFluids) {
+            val predicate = segments[segment]!!.predicate
+            if (predicate != null && !predicate.test(fluid)) {
+                continue
+            }
+
+            val requesters = requestedFluids[fluid]?.toMutableSet()
+            if (requesters.isNullOrEmpty()) {
+                continue
+            }
+
+            // Take fluids on a first-come-first-serve basis from suppliers
+            val totalRequested = min(requesters.sumOf { it.amount }, segments[segment]!!.fluidPerTick)
+            if (totalRequested == 0L) {
+                continue
+            }
+            var totalSupplied = 0L
+            for (supplier in suppliers) {
+                val remaining = totalRequested - totalSupplied
+                if (remaining == 0L) {
+                    break
+                }
+
+                val toTake = min(remaining, supplier.amount)
+                try {
+                    supplier.block.removeFluid(supplier.name, supplier.fluid, toTake)
+                } catch (e: Throwable) {
+                    e.printStackTrace()
+                    continue
+                }
+                totalSupplied += toTake
+            }
+
+            // Round-robin distribute to requesters
+            while (totalSupplied != 0L) {
+                val maxFluidPerRequester = if (totalSupplied / requesters.size != 0L) {
+                    totalSupplied / requesters.size
+                } else {
+                    // eg: splitting 3 mB amongst 5 requesters
+                    1
+                }
+                val iterator = requesters.iterator()
+                while (iterator.hasNext() && totalSupplied != 0L) {
+                    val requester = iterator.next()
+                    if (requester.amount < maxFluidPerRequester) {
+                        requester.block.addFluid(requester.name, requester.fluid, requester.amount)
+                        iterator.remove()
+                        totalSupplied -= requester.amount
+                    } else {
+                        requester.block.addFluid(requester.name, requester.fluid, maxFluidPerRequester)
+                        totalSupplied -= maxFluidPerRequester
+                    }
+                }
+            }
+
+            // Only allow one type of fluid to be distributed per tick
+            break
+        }
+    }
+
+    private fun startTicker(segment: UUID) {
+        check(segment !in tickers) { "Ticker already active" }
+
+        val dispatcher = pluginInstance.minecraftDispatcher
+        tickers[segment] = pluginInstance.launch(dispatcher) {
+            while (true) {
+                delay(PylonConfig.fluidIntervalTicks.toLong())
```

Comment:
```suggestion
                delay(PylonConfig.fluidIntervalTicks.ticks)
```
`delay` uses milliseconds, `.ticks` converts ticks to milliseconds